scripts/helper/functions.sh

```diff
@@ -277,7 +277,7 @@ mds_login()
   OUTPUT=$(
   expect <<END
     log_user 1
-    spawn confluent login --ca-cert-path /etc/kafka/secrets/snakeoil-ca-1.crt --url $MDS_URL
+    spawn confluent-v1 login --ca-cert-path /etc/kafka/secrets/snakeoil-ca-1.crt --url $MDS_URL -vvvv
```

Comment:
@brianstrauch I looked into this issue, and it works with `-vvvv` but fails without `-vvvv`.   This is a regression and results in a lot of debug output that is distracting to the demo.  Any ideas on what's happening here?

Reply:
In one of the later 1.x versions of the CLI (somewhere around 1.30 iirc) we started hiding the "login success" message (not sure why that was, this happened before I joined). I can experiment with less verbosity (maybe `-vv` or `-vvv`) so we can strike a good balance where the demo works but the output isn't distracting.

Reply:
Unfortunately `-vv` prints nothing and `-vvv` prints too much. How about we print a generic "Logged in to the Confluent CLI" message if login succeeds, and the full output only if it fails?

Reply:
It should work without any verbose flags. Perhaps the script can be modified to check error codes:

```
root@tools:/# confluent-v1 login --ca-cert-path /etc/kafka/secrets/snakeoil-ca-1.crt --url $MDS_URL
Enter your Confluent credentials:
Username: kjdf
Password: ****
Error: metadata service backend error: 401 Unauthorized: {
"servlet":"default",
"message":"Unauthorized",
"url":"/security/1.0/authenticate",
"status":"401"
}
root@tools:/# echo $?
1
root@tools:/# confluent-v1 login --ca-cert-path /etc/kafka/secrets/snakeoil-ca-1.crt --url $MDS_URL
Enter your Confluent credentials:
Username: superUser
Password: *********
root@tools:/# echo $?
0
```

Reply:
Or search for errors in output

Reply:
@brianstrauch WDYT about this solution?


```
[12:55:51] ~/git/cp-demo/scripts(PR-404) ✗: git diff
diff --git a/scripts/helper/functions.sh b/scripts/helper/functions.sh
index 7811c4e..b2ad77e 100755
--- a/scripts/helper/functions.sh
+++ b/scripts/helper/functions.sh
@@ -281,12 +281,13 @@ mds_login()
     expect "Username: "
     send "${SUPER_USER}\r";
     expect "Password: "
+    log_user 0
     send "${SUPER_USER_PASSWORD}\r";
+    log_user 1
     expect "Logged in as "
     set result $expect_out(buffer)
 END
   )
-  echo "$OUTPUT"
   if [[ ! "$OUTPUT" =~ "Logged in as" ]]; then
     echo "Failed to log into MDS. Please check all parameters and run again."
     exit 1
```

Reply:
My favorite solution of the two is to check for "Error:" in the output.

Reply:
Checking for error codes seems to be the cleanest approach here!

Reply:
@brianstrauch can you please validate the code below (it works in my env) and then drop/replace into your PR?

```
mds_login()
{
  MDS_URL=$1
  SUPER_USER=$2
  SUPER_USER_PASSWORD=$3

  # Log into MDS
  if [[ $(type expect 2>&1) =~ "not found" ]]; then
    echo "'expect' is not found. Install 'expect' and try again"
    exit 1
  fi
  echo -e "\n# Login to MDS using Confluent CLI"
  OUTPUT=$(
  expect <<END
    log_user 1
    spawn confluent-v1 login --ca-cert-path /etc/kafka/secrets/snakeoil-ca-1.crt --url $MDS_URL
    expect "Username: "
    send "${SUPER_USER}\r";
    expect "Password: "
    send "${SUPER_USER_PASSWORD}\r";
    expect EOF
    catch wait result
    exit [lindex \$result 3]
END
  )
  if [ $? -ne 0 ]; then
    echo "Failed to log into MDS. Please check all parameters and run again."
    exit
  fi
}
```

Reply:
Worked perfectly- updated!